src/test/java/hudson/tasks/junit/CaseResultTest.java

```diff
@@ -98,24 +98,23 @@ public void testIssue20090516() throws Exception {
         // piggy back tests for annotate methods
         assertOutput(cr,"plain text", "plain text");
         assertOutput(cr,"line #1\nhttp://nowhere.net/\nline #2\n",
-                     "line #1\nhttp://nowhere.net/\nline #2\n");
+                     "line #1\n" + "<a href=\"http://nowhere.net/\">http://nowhere.net/</a>\n" + "line #2\n");
         assertOutput(cr,"failed; see http://nowhere.net/",
-                     "failed; see http://nowhere.net/");
+                     "failed; see <a href=\"http://nowhere.net/\">http://nowhere.net/</a>");
         assertOutput(cr,"failed (see http://nowhere.net/)",
-                     "failed (see http://nowhere.net/)");
+                     "failed (see <a href=\"http://nowhere.net/\">http://nowhere.net/</a>)");
         assertOutput(cr,"http://nowhere.net/ - failed: http://elsewhere.net/",
-                     "http://nowhere.net/ - failed: " +
-                     "http://elsewhere.net/");
+                     "<a href=\"http://nowhere.net/\">http://nowhere.net/</a> - failed: <a href=\"http://elsewhere.net/\">http://elsewhere.net/</a>");
         assertOutput(cr,"https://nowhere.net/",
-                     "https://nowhere.net/");
+                     "<a href=\"https://nowhere.net/\">https://nowhere.net/</a>");
         assertOutput(cr,"stuffhttp://nowhere.net/", "stuffhttp://nowhere.net/");
         assertOutput(cr,"a < b && c < d", "a &lt; b &amp;&amp; c &lt; d");
         assertOutput(cr,"see <http://nowhere.net/>",
-                     "see &lt;http://nowhere.net/>");
+                     "see &lt;<a href=\"http://nowhere.net/\">http://nowhere.net/</a>&gt;");
         assertOutput(cr,"http://google.com/?q=stuff&lang=en",
-                     "http://google.com/?q=stuff&amp;lang=en");
+                     "<a href=\"http://google.com/?q&#61;stuff&amp;lang&#61;en\">http://google.com/?q&#61;stuff&amp;lang&#61;en</a>");
```

Comment:
this looks questionable - should it not be `<a href=\"http://google.com/?q=stuff&lang=en\">http://google.com/?q&#61;stuff&amp;lang&#61;en</a>

Reply:
Equals signs are getting escaped, we could easily put them back in after it has been sanitised with a replaceAll but then we are kind of undoing the sanitising 😕